src/main/java/com/datasonnet/plugins/CSVFormatPlugin.java

```diff
@@ -27,7 +30,7 @@ public CSVFormatPlugin() {
 
     }
 
-    public Value read(String input, Map<String, Object> params) throws IOException {
+    public Value read(Object input, Map<String, Object> params) throws PluginException {
```

Comment:
I think we can make plugins _also_ parametrized by T instead of giving them generic objects for input and writing unparametrized Document types, which should help sort some of our types

Reply:
Also, just noticing this now, we should use `@Override`

Reply:
I actually tried that, but scala was complaining. And the reasons are deep in the sjsonnet and ujson, so I couldn't do much there...